nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -326,6 +329,27 @@ public void process(InputStream in) throws IOException {
         return splitFlowFiles;
     }
 
+    /**
+     * Will merge the two files. This operation has different semantics then
+     * sessioin.merge. Also, it always expects 2 and no more then 2 flow files,
+     * so vararg signature primarily for simplification.
+     */
+    private FlowFile merge(ProcessSession session, FlowFile... flowFiles) {
+        FlowFile mergedFlowFile = session.create();
```

Comment:
Please pass the source FlowFile to session.create so that it generates fork provenance event. Calling create() without that loses track to the parent provenance.